### PR TITLE
Arrays

### PR DESCRIPTION
**Pull request by Punicorn**
Interesting. "unique" doesn`t exactly works with forEach method.
P.S. Sorry for delay. I was sick about a week.

### CODE REVIEW

File: 2.Arrays/arrays.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+function isArray1(arr) {
+	return arr instanceof Array;
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: 2 spaces

File: 2.Arrays/arrays.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+console.log(isArray1([3,null,undefined,4]));
+
+function isArray2(arr) {
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: this function doesn't make any sense

File: 2.Arrays/arrays.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	return Array.isArray(arr);
+}
+
+console.log(isArray2([3,null,undefined,4]));
+console.log(isArray2({0:2, 1:3}));
+
+function range(from, to, step) {
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: cases should work:
range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) [] 
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5)
currently I have infinity loop for some cases

File: 2.Arrays/arrays.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	return(arr);
+}
+
+console.log(range(1,10,2));
+
+function compactCycle(arr) {
+	var newArr = [];
+	for (let i = 0; i < arr.length; i++) {
+		(arr[i] == true) && (newArr.push(arr[i]));
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: 1. brackets are not required here
2.  `arr[i] && newArr.push(arr[i])` should work

File: 2.Arrays/arrays.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	return newArr;
+}
+
+console.log(compactCycle(['', 'true', 1, 0, null]));
+
+function compact(arr) {
+	var newArr = [];
+	newArr = arr.filter(function(x) {
+		return x == true;
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: `return x`?

File: 2.Arrays/arrays.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+console.log(unique([1,1,2,3,3,3,4]));
+
+function last(arr) {
+	return arr[arr.length - 1];
+}
+
+console.log(last([2,5,48]));
+
+function excludeLast(arr, n) {
+	(n === undefined)  && (n = 1);
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: you have 2 spaces after the 1 condition

File: 2.Arrays/arrays.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	return arr.slice(0, -n);
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: when n is negative value like -5 you will have arr + [undef, undef, undef, undef, undef]

File: 2.Arrays/index.html [hunk 8]
```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Arrays</title>
+  </head>
+  <body>
+  	<header>
```
Comment (NikitaSedyx) [inline 2.Arrays/index.html]: 2 spaces for html too

File: 2.Arrays/index.html [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+  	<nav>
+  		Nav
+  		<ul>
+  			<li>First</li>
+  			<li>Second</li>
+  		</ul>
+  	</nav>
+
+  	<article>
```
Comment (NikitaSedyx) [inline 2.Arrays/index.html]: docs suggests to use headers inside articles and sections

File: 2.Arrays/index.html [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  	Aside
+  		<nav>
+  			<h1>First nav</h1>
+  		</nav>
+  		<nav>
+  			Second nav
+  		</nav>
+  	</aside>
+
+  	<main>
```
Comment (NikitaSedyx) [inline 2.Arrays/index.html]: imo it's a bit strange you have articles outside main content

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

File: 2.Arrays/arrays.js (hunk 2 above)
Comment (Punicorn) [reply 2.Arrays/arrays.js]: Which of these? I`ve found both in "JavaScript. The Definitive Guide" by David Flanagan, chapter 7.10.

Comment (NikitaSedyx) [reply 2.Arrays/arrays.js]: you just wrapped standard function, there is no sense in this action

File: 2.Arrays/arrays.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
+function isArray1(arr) {
+  return Object.prototype.toString.call(arr) === '[object Array]';
+}
+
+console.log(isArray1([3,null,undefined,4]));
+console.log(isArray1(null));
+
+function isArray2(arr) {
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: don't see it has been changed. please look at previous comment

File: 2.Arrays/arrays.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+  if (!to) {
+    to = from;
+    from = 0;
+  }
+  !step && (step = 1);
+  for(let i = from; (i < to) == (step > 0); i += step) {
+    if (i == to) continue; // drops the last num in negative case
+    arr.push(i);
+  }
+  return(arr);
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: `return arr`? it will work without brackets

File: 2.Arrays/arrays.js [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+function isArray2(arr) {
+  return Array.isArray(arr);
+}
+
+console.log(isArray2([3,null,undefined,4]));
+console.log(isArray2({0:2, 1:3}));
+
+function range(from, to, step) {
+  var arr = [];
+  if (!to) {
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: `to` can be 0, !0 is false

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]
basically looks good, please several comments

File: 2.Arrays/arrays.js [hunk 14]
```diff
@@ -0,0 +1,102 @@
+console.log(
+  Object.prototype.toString.call([3,'da',4]) === '[object Array]');
+console.log([48] instanceof Array);
+
+function isArray2(arr) {
+  return Array.isArray(arr);
```
Comment (NikitaSedyx) [inline 2.Arrays/arrays.js]: this should be removed. and you should have 2 functions with return instanceof and Object.prototype.toString.call

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**